game_logic/src/lib.rs

```diff
@@ -1,18 +1,19 @@
-#[derive(Copy)]
-pub struct LogicState {
+pub trait GameEvent {
+  fn works(&self) -> bool;
+}
+
+pub struct LogicState<'a> {
   good: bool,
   edge_count: usize,
-  x: f32,
-  y: f32
+  events: Vec<&'a GameEvent>
```

Comment:
This makes a Vec of Trait Objects (dynamically dispatched type-erased objects), which is different than a Vec of some type which implements GameEvent like 
```rust
pub struct LogicState<'a, T> where T: GameEvent {
  ...
  events: Vec<&'a T>
}
```
Is that what you intended?